stats/cloud/collector.go

```diff
@@ -465,6 +494,11 @@ func (c *Collector) pushMetrics() {
 			logrus.WithFields(logrus.Fields{
 				"error": err,
 			}).Warn("Failed to send metrics to cloud")
+			if c.shouldStopSendingMetrics(err) {
+				logrus.Warn("Stop sending metrics to cloud due to error.")
```

Comment:
Minor typo, the error should be in the past tense, something like `Stopped sending metrics to the cloud`. Also, this error could probably be tied together with the error above, so we only ever emit one of them - either the normal one, or the stopping one, something like this:
```go
if c.shouldStopSendingMetrics(err) {
	logrus.WithError(err).Warn("Stopped sending metrics to the cloud due to an error")
} else {
	logrus.WithError(err).Warn("Failed to send metrics to the cloud")
}
```